src/buildApplication/generateFunctionsMap.ts

```diff
@@ -0,0 +1,314 @@
+
+	await Promise.all(
+		files.map(async file => {
+			const filepath = join(dir, file);
+			const isDirectory = (await stat(filepath)).isDirectory();
+			if (isDirectory) {
+				const dirResultsPromise = file.endsWith('.func')
+					? processFuncDirectory(setup, file, filepath)
+					: processDirectoryRecursively(setup, filepath);
+				const dirResults = await dirResultsPromise;
+				dirResults.invalidFunctions?.forEach(fn => invalidFunctions.add(fn));
+				dirResults.functionsMap?.forEach((value, key) =>
+					functionsMap.set(key, value)
+				);
+				dirResults.webpackChunks?.forEach((value, key) =>
+					webpackChunks.set(key, value)
+				);
+			}
+		})
+	);
+
+	return {
+		invalidFunctions,
+		functionsMap,
+		webpackChunks,
+	};
+}
+
+async function processFuncDirectory(
+	setup: ProcessingSetup,
+	file: string,
+	filepath: string
+): Promise<Partial<DirectoryProcessingResults>> {
+	const relativePath = relative(setup.functionsDir, filepath);
+
+	const functionConfigFile = join(filepath, '.vc-config.json');
+	let functionConfig: { runtime: 'edge'; entrypoint: string } | null = null;
+	try {
+		const contents = await readFile(functionConfigFile, 'utf8');
+		functionConfig = JSON.parse(contents);
+	} catch {
+		functionConfig = null;
+	}
+
+	if (functionConfig?.runtime !== 'edge') {
+		return {
+			invalidFunctions: new Set([file]),
+		};
+	}
+
+	const functionFile = join(filepath, functionConfig.entrypoint);
+	let functionFileExists = false;
+	try {
+		await stat(functionFile);
+		functionFileExists = true;
+	} catch {
+		/* empty */
+	}
+
+	if (!functionFileExists) {
+		return {
+			invalidFunctions: new Set([file]),
+		};
+	}
+
+	const functionsMap = new Map<string, string>();
+	const webpackChunks = new Map<number, string>();
+
+	let contents = await readFile(functionFile, 'utf8');
+	contents = fixFunctionContents(contents);
+
+	if (setup.experimentalMinify) {
+		const { updatedFunctionContents, extractedWebpackChunks } =
+			extractWebpackChunks(setup.tmpWebpackDir, contents, webpackChunks);
+		contents = updatedFunctionContents;
+		extractedWebpackChunks.forEach((value, key) =>
+			webpackChunks.set(key, value)
+		);
+	}
+
+	const newFilePath = join(setup.tmpFunctionsDir, `${relativePath}.js`);
+	await mkdir(dirname(newFilePath), { recursive: true });
+	await writeFile(newFilePath, contents);
+
+	functionsMap.set(
+		normalizePath(
+			relative(setup.functionsDir, filepath).slice(0, -'.func'.length)
+		),
+		normalizePath(newFilePath)
+	);
+
+	return {
+		functionsMap,
+		webpackChunks,
+	};
+}
+
+/**
+ * Fixes the function contents in miscellaneous ways.
+ *
+ * Note: this function contains hacks which are quite brittle and should be improved ASAP.
+ *
+ * @param contents the original function's file contents
+ * @returns the updated/fixed contents
+ */
+function fixFunctionContents(contents: string) {
+	contents = contents.replace(
+		// TODO: This hack is not good. We should replace this with something less brittle ASAP
```

Comment:
@GregBrimble here I just added a url to the source code, but I don't 100% remember the reasoning for this?

Could you remind me so that I can improve the comment?